lib/hammer_cli_foreman/api/connection.rb

```diff
@@ -36,24 +37,34 @@ def create_authenticator(uri, settings)
         if ssl_cert_authentication?(settings) && !use_basic_auth?(settings)
           @authenticator = VoidAuth.new(_('Using certificate authentication.'))
         else
-          if settings.get(:foreman, :use_sessions)
-            @authenticator = InteractiveBasicAuth.new(
-              settings.get(:_params, :username) || ENV['FOREMAN_USERNAME'],
-              settings.get(:_params, :password) || ENV['FOREMAN_PASSWORD']
-            )
-            @authenticator = SessionAuthenticatorWrapper.new(@authenticator, uri)
-          else
-            username = settings.get(:_params, :username) || ENV['FOREMAN_USERNAME'] || settings.get(:foreman, :username)
-            password = settings.get(:_params, :password) || ENV['FOREMAN_PASSWORD']
-            if (password.nil? && (username == settings.get(:foreman, :username)))
-              password = settings.get(:foreman, :password)
-            end
-            @authenticator = InteractiveBasicAuth.new(username, password)
-          end
-          @authenticator
+          @authenticator = InteractiveTokenAuth.new(settings.get(:_params, :token) || ENV['KEYCLOAK_TOKEN'])
```

Comment:
@mbacovsky I am trying to create an object of the class `InteractiveTokenAuth` but i am failing in doing so. Can you please take a look?

Reply:
The whole idea is to set the @token in the `@authenticator` so that it can be passed to foreman api.